awx_collection/plugins/modules/tower_group.py

```diff
@@ -104,87 +66,62 @@
-        source_regions=dict(),
-        source_vars=dict(),
-        instance_filters=dict(),
-        group_by=dict(),
-        source_script=dict(),
-        overwrite=dict(type='bool'),
-        overwrite_vars=dict(type='bool'),
-        update_on_launch=dict(type='bool'),
         state=dict(choices=['present', 'absent'], default='present'),
     )
 
+    # Create a module for ourselves
     module = TowerModule(argument_spec=argument_spec, supports_check_mode=True)
 
+    # Extract our parameters
     name = module.params.get('name')
+    new_name = module.params.get('new_name')
     inventory = module.params.get('inventory')
-    credential = module.params.get('credential')
+    description = module.params.get('description')
     state = module.params.pop('state')
-
     variables = module.params.get('variables')
-    if variables:
-        if variables.startswith('@'):
-            filename = os.path.expanduser(variables[1:])
-            with open(filename, 'r') as f:
-                variables = f.read()
-
-    json_output = {'group': name, 'state': state}
-
-    tower_auth = tower_auth_config(module)
-    with settings.runtime_values(**tower_auth):
-        tower_check_mode(module)
-        group = tower_cli.get_resource('group')
-        try:
-            params = module.params.copy()
-            params['create_on_missing'] = True
-            params['variables'] = variables
-
-            inv_res = tower_cli.get_resource('inventory')
-            inv = inv_res.get(name=inventory)
-            params['inventory'] = inv['id']
-
-            if credential:
-                cred_res = tower_cli.get_resource('credential')
-                cred = cred_res.get(name=credential)
-                params['credential'] = cred['id']
-
-            if state == 'present':
-                result = group.modify(**params)
-                json_output['id'] = result['id']
-            elif state == 'absent':
-                result = group.delete(**params)
-        except (exc.NotFound) as excinfo:
-            module.fail_json(msg='Failed to update the group, inventory not found: {0}'.format(excinfo), changed=False)
-        except (exc.ConnectionError, exc.BadRequest, exc.AuthError) as excinfo:
-            module.fail_json(msg='Failed to update the group: {0}'.format(excinfo), changed=False)
-
-    json_output['changed'] = result['changed']
-    module.exit_json(**json_output)
 
+    # Attempt to lookup the related items the user specified (these will fail the module if not found)
+    inventory_id = module.resolve_name_to_id('inventories', inventory)
+
+    # Attempt to lookup team based on the provided name and org ID
+    group = module.get_one('groups', **{
+        'data': {
+            'name': name,
+            'inventory': inventory_id
+        }
+    })
+
+    # If the variables were spacified as a file, load them
+    if variables:
+        variables = module.load_variables_if_file_specified(variables, 'variables')
```

Comment:
I don't like that we're still doing this, but I don't want to add one more deprecated thing into this PR.

But I might make an issue to get rid of all these instances, and wanted to ask how you would feel about that. In order to optionally take the "@" character, that means the parameter type is string, and I would much prefer that the type be a dict.

Reply:
So I almost said I 100% agree on this but then I started to think about it....
The only reason this notation was supported was because it had to be fed into Tower CLI that way. Since we are deprecating Tower CLI with this PR one could argue that now is the perfect time to drop this. If we wanted to do that we could provide in the release notes something like:

IF you used the @ method to import a vars file, here is how you would convert your code to have Ansible load a vars file.... one of the caveats would be that the @ syntax as is loads the file from the targeted node where as using a lookup plugin would load from the controller.

Also, as an FYI, in my test I passed in something like:
   variables:
       foo: bar
Ansible generates a warning that its forcing a dict into a str but other than that it works.

Reply:
> IF you used the @ method to import a vars file, here is how you would convert your code to have Ansible load a vars file....

probably similar to:

https://github.com/ansible/awx/blob/devel/awx_collection/plugins/modules/tower_job_template.py#L242

I'm cautiously optimistic that no one will have problems with these parameter type changes  - because the change was prompted by Ansible core itself. Their CI started enforcing types, but old modules were given an exception. The variables param here technically passes their CI, but doesn't respect the _spirit_ of enforcing types. So at minimum, we have someone else to point the finger at.

Reply:
We can go ahead and take this out (changes currently in progress).  Do we want to raise an exception or `warn` that the "@" syntax is no longer supported?